src/cli/download_handler.rs

```diff
@@ -84,15 +97,65 @@ impl DownloadHandler {
     }
 
     pub fn run(&self) -> HandlerResult {
+        let destination = self.selection_destination_for_install()?;
+        self.destination_may_not_dir(&destination)?;
         let github = GithubClient::from_environment();
         let release = self.fetch_release(&github)?;
         let selected_asset = self.select_asset(release)?;
+        self.asset_may_not_be_archive(&selected_asset.name)?;
```

Comment:
Not needed here. This is already done when you call `installer::install`.
Remember to delete the `asset_may_not_be_archive` function.